src/View/Form/DocumentContext.php

```diff
@@ -421,7 +421,7 @@ public function error(string $field): array
             $errors = $entity->getError(array_pop($parts));
         }
 
-        if (!$errors && $entityErrors && !is_array($entity)) {
+        if (!$errors && $entityErrors) {
```

Comment:
Looking good, we'll need to get some tests added before we can merge this though.  This class has some existing tests https://github.com/cakephp/elastic-search/blob/4.x/tests/TestCase/View/Form/DocumentContextTest.php

I'm happy to help if you need a hand.

Reply:
I am unable to set up the test environment, so adding test code is difficult. Could you please assist me with this?

I believe the issue can be reproduced by specifying the following data in `new Article` and triggering the corresponding error:

```json
{
  "name": {
    "ja": [
      {
        "family_name": "XX",
        "given_name": "XX"
      }
    ],
    "en": [
      {
        "family_name": "XX",
        "given_name": "XXX"
      }
    ]
  }
}
```